qa/tasks/vip.py

```diff
@@ -0,0 +1,171 @@
+import contextlib
+import ipaddress
+import logging
+import re
+
+from teuthology.config import config as teuth_config
+
+log = logging.getLogger(__name__)
+
+
+def subst_vip(ctx, cmd):
+    p = re.compile(r'({{VIP(\d+)}})')
+    for m in p.findall(cmd):
+        n = int(m[1])
+        if n >= len(ctx.vip["vips"]):
+            log.warning(f'no VIP{n} (we have {len(ctx.vip["vips"])})')
+        else:
+            cmd = cmd.replace(m[0], str(ctx.vip["vips"][n]))
+
+    if '{{VIPPREFIXLEN}}' in cmd:
```

Comment:
people will think that `{{VIPPREFIXLEN}}` etc is a real jinja2 template. But I think this is ok for now.